src/localization/database/cost_matrix.cpp

```diff
@@ -0,0 +1,94 @@
+#include "cost_matrix.h"
+#include <fstream>
```

Comment:
It is a good practice to have the following order of includes:
1. your project includes
2. external library includes
3. standard headers

This way you minimize the risk of includes "contamination". For example, if your include needs `std::string` and in your whole codebase it will appear _after_ `#include <string>` in other files, you will miss the fact that it has a missing include. Hope this makes sense.